CMakeLists.txt

```diff
@@ -5,6 +5,24 @@
 
 cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
 
+option(MBED_BUILD_UNITTESTS "Run unit tests only." OFF)
+
+if(MBED_BUILD_UNITTESTS)
+    project(mbed-unittests)
```

Comment:
Should we be defining a project just for the unittests? Does it make sense to always create an mbed-os project instead? There are probably other cases where users may want to build from mbed-os root. It may also be useful to set the current mbed-os version so users can depend on specific versions of mbed.

Reply:
As suggested we created a new mbed-os project